### PR TITLE
Integrate new AINDLickDetector

### PR DESCRIPTION
**Pull request by XX-Yin**
### Pull Request instructions:
- Please follow the [update protocol](https://github.com/AllenNeuralDynamics/aind-behavior-blog/wiki/Software-Update-Procedures)
- Answer the questions below in detail. Your responses will be emailed to experimenters. 
- If the experimenters must do anything new, provide detailed step by step instructions on the [wiki](https://github.com/AllenNeuralDynamics/aind-behavior-blog/wiki)
- Use markdown syntax in order for your comments to be rendered reliably in the email: "1." instead of "1)", use four spaces for indents.
- If you use the keyword "skip email" in the title, it will skip the email updates
- Merges from "develop" into "production_testing" should use the keyword "production merge" in the title for reliable indexing of updates
- Merges from "production_testing" into "main" should use the keyword "update main"
  
### Describe changes:
1. The new lick detector was integrated into the bonsai workflow. 
2. Updated the Settings_box#.csv
image
image
image
image
### What issues or discussions does this update address?
The new lick detector and Janelia lick detector exist in different bonsai workflows, which makes automatic update difficult.

### Describe the expected change in behavior from the perspective of the experimenter
1. A new field called `AINDLickDetector` should be added to `Settings_box#.csv`. `0`, using the Janelia lick detector. `1`, using the new AINDLickDetector. 
2. COM ports (`LeftLickDetector` and `RightLickDetector`) should be set in the `Settings_box#.csv` if we want to use the AINDLickDetector. 
3. **NOTE TO USERS:** This PR updates some bonsai packages, the first time the GUI runs bonsai wont open because things are installing behind the scene. So close the GUI and open it again

### Describe the outcome of testing this update on a rig in 447

Tested in Ephys rig 1 and 3.

### CODE REVIEW

**Comment by alexpiet** [comment]
Lets call it something other than `NewLickDetector` since that won't be informative in the future. How about `AINDLickDetector`?

File: bonsai/Bonsai.config [hunk 1]
```diff
@@ -4,8 +4,9 @@
     <Package id="AForge" version="2.2.5" />
     <Package id="AForge.Video" version="2.2.5" />
     <Package id="AForge.Video.DirectShow" version="2.2.5" />
+    <Package id="AllenNeuralDynamics.LicketySplitLickDetector" version="0.1.0-preview2024020101" />
     <Package id="AssimpNet" version="4.1.0" />
-    <Package id="Bonsai" version="2.7.2" />
+    <Package id="Bonsai" version="2.8.0" />
```
Comment (alexpiet) [inline bonsai/Bonsai.config:9]: Do we have to manually install the updated bonsai version on each computer?

**Review by alexpiet — CHANGES_REQUESTED** [review]

**Comment by XX-Yin** [comment]
> AINDLickDetector

That sounds good!

File: bonsai/Bonsai.config (hunk 1 above)
Comment (XX-Yin) [reply bonsai/Bonsai.config:9]: No, we don't. It will be automatically updated when we open the bonsai (or the bonsai workflow).

**Comment by XX-Yin** [comment]
> Lets call it something other than `NewLickDetector` since that won't be informative in the future. How about `AINDLickDetector`?

I have changed the name

File: src/setting_templates/Settings_box1.csv [hunk 2]
```diff
@@ -7,3 +7,5 @@ BonsaiOsc4,4005
 HighSpeedCamera,0
 SideCamera,12345678
 BottomCamera,12345678
+AINDLickDetector,0
+,
```
Comment (alexpiet) [inline src/setting_templates/Settings_box1.csv:11]: remove the comma on the blank line

**Review by alexpiet — CHANGES_REQUESTED** [review]

**Comment by alexpiet** [comment]
I tested on my laptop and it looks good. Lets merge after fixing the stray comma mentioned above. I'll update the settings files on the 447 computers today or tomorrow

**Comment by alexpiet** [comment]
When I tested it on my laptop, the bonsai updates take a while to run, so I made a note on the PR for the RAs to reopen the GUI if bonsai doesnt open

File: src/setting_templates/Settings_box1.csv (hunk 2 above)
Comment (alexpiet) [reply src/setting_templates/Settings_box1.csv]: fixed it

**Review by alexpiet — APPROVED** [review]

**Comment by alexpiet** [comment]
I updated the settings files on the 447 computers

**Pull request merged**